cpp/core/operators/shuffle/SplitterBase.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "operators/shuffle/type.h"
+
+namespace gluten {
+
+class SplitterBase {
+ public:
+  /**
+   * Spill for fixed size of partition data
+   */
+  virtual arrow::Status SpillFixedSize(int64_t size, int64_t* actual) = 0;
+
+  virtual arrow::Status Split(ColumnarBatch* cb) = 0;
+
+  virtual arrow::Status Stop() = 0;
+
+  int64_t TotalBytesWritten() const {
+    return total_bytes_written_;
+  }
+
+  int64_t TotalBytesSpilled() const {
```

Comment:
If it is a Splitter using RSS, generally there will be no Shuffle Spill, so this part can be left to the subclass definition.